Cesium.CodeGen/Ir/Expressions/SubscriptingExpression.cs

```diff
@@ -35,7 +35,12 @@ public IExpression Lower()
     public IValue Resolve(IDeclarationScope scope)
     {
         if (_expression is not IdentifierExpression identifier)
```

Comment:
`IdentifierExpression` is also an `IValueExpression`, too, so you may remove the first type check.

Reply:
But if we dont have the first check, method will always return `IValueExpresssion.Resolve(scope);` and not the `LValueArrayElement`.

Reply:
Ah, you're right. Let's clarify the logic a bit then?
Something like
```csharp
switch (_expression)
{
  case IdentifierExpression identifier: …
  case IValueExpression valueExpression: …
  default: throw new CompilationException(…);
}
```

Reply:
Yep. Thats looks better